synapse/cores/common.py

```diff
@@ -1901,6 +1901,31 @@ def formTufoByTufo(self, tufo):
 
         return self.formTufoByProp(form, valu, **props)
 
+    def formTufosByItems(self, items):
+        '''
+        Forms tufos by prop, given a tuple of (form, valu, props) tuples.
+
+        Args:
+
+            items (tuple): A tuple of tuples of (form, valu, props)
+
+        Returns: tuple of tufos and errors # FIXME not sure how to express this in docs format
+
+        Example:
+
+            items = ( ('foo:thing', 'hehe', {'a': 1}), ('bar:thing', 'haha', {'b': 2}), )
+            results = core.formTufoByItems(items)
+        '''
+        retval = []
+
+        for form, valu, props in items:
```

Comment:
the xact object is cached by the storage layer per thread, so the xact created in this will be returned by formTufoByProp's xact call.